src/librustc/session/mod.rs

```diff
@@ -598,8 +598,12 @@ impl Session {
     }
 
     pub fn must_not_eliminate_frame_pointers(&self) -> bool {
-        self.opts.debuginfo != DebugInfoLevel::NoDebugInfo ||
-        !self.target.target.options.eliminate_frame_pointer
+        if let Some(x) = self.opts.cg.force_frame_pointers {
+            x
+        } else {
+            !self.target.target.options.eliminate_frame_pointer ||
+            self.opts.debuginfo != DebugInfoLevel::NoDebugInfo
```

Comment:
Isn't the point of this PR to not tie frame pointer emission to the debuginfo setting?

Reply:
This keeps it by default as suggested by the linked issue:

Reply:
Most of the comments in #11906 ask that -g not affect debuginfo, e.g.: https://github.com/rust-lang/rust/issues/11906#issuecomment-269022609, https://github.com/rust-lang/rust/issues/11906#issuecomment-294662037, https://github.com/rust-lang/rust/issues/11906#issuecomment-353080693.

#48785 disagrees, though maybe leaves the door open to have it be a target setting.

I tend to think it's best if the presence or absence of debuginfo does not affect code generation.

Reply:
Why did we link the two anyway, intitially? Was it something in the way we made debuginfo which is no longer relevant?

Reply:
> Why did we link the two anyway, intitially? Was it something in the way we made debuginfo which is no longer relevant?

Yes, @dotdash fixed things in our LLVM IR generation so that now debuggers can handle function arguments without frame pointers being present. Before, debuggers had problems displaying variables under certain conditions.